gh-104469: Update README.txt for _testcapi (#104529)
python/cpython: Modules/_testcapi/README.txt

Tests in this directory are compiled into the _testcapi extension.
The main file for the extension is Modules/_testcapimodule.c, which
calls `_PyTestCapi_Init_*` from these functions.

General guideline when writing test code for C API.

> * Use clinic tool as possible.

Reviewer: ```suggestion
* Use Argument Clinic to minimise the amount of boilerplate code.
```

> * Add a newline between the the argument spec and the docstring.

Reviewer 2: ```suggestion
* Add a newline between the argument spec and the docstring.
```

> * If a test decription is needed, make sure the added docstring clearly and succinctly describes purpose of the function.

Reviewer: ```suggestion
* If a test description is needed, make sure the added docstring clearly and succinctly describes purpose of the function.
```

* DRY, use the clone feature of Argument Clinic.

> * Try to avoid adding new interned strings; reuse existing parameter names if possible and use the `as` feature to override the C name.

Reviewer: ```suggestion
* Try to avoid adding new interned strings; reuse existing parameter names if possible. Use the `as` feature of Argument Clinic to override the C variable name, if needed.
```
